x-pack/platform/plugins/shared/streams/server/lib/streams/create_streams_global_search_result_provider.ts

```diff
@@ -0,0 +1,191 @@
+        takeUntil(aborted$)
+      );
+    },
+  };
+}
+
+async function findStreams({
+  term,
+  types,
+  maxResults,
+  storageClient,
+  client,
+  core,
+}: {
+  term: string;
+  types: string[];
+  maxResults: number;
+  storageClient: StreamsStorageClient;
+  client: IScopedClusterClient;
+  core: CoreSetup;
+}) {
+  const [coreStart] = await core.getStartServices();
+  const soClient = coreStart.savedObjects.getUnsafeInternalClient();
+  const uiSettingsClient = coreStart.uiSettings.asScopedToClient(soClient);
+  const groupStreamsEnabled = await uiSettingsClient.get('observability:streamsEnableGroupStreams');
+
+  // This does NOT included unmanaged Classic streams
+  const searchResponse = await storageClient.search({
+    size: maxResults,
+    track_total_hits: false,
+    query: {
+      bool: {
+        should: [
+          {
+            wildcard: {
+              name: {
+                value: `*${term}*`,
+              },
+            },
+          },
+          {
+            match: {
+              description: term,
+            },
+          },
+        ],
+        minimum_should_match: 1,
+      },
+    },
+  });
+
+  const privileges = await checkAccessBulk({
+    names: searchResponse.hits.hits
+      .filter((hit) => !Streams.GroupStream.Definition.is(hit._source))
+      .map((hit) => hit._source.name),
+    scopedClusterClient: client,
+  });
+
+  const hitsWithAccess = searchResponse.hits.hits.filter((hit) => {
+    if (Streams.GroupStream.Definition.is(hit._source)) return true;
+    return privileges[hit._source.name]?.read === true;
+  });
+
+  if (types.length === 0) {
+    return hitsWithAccess.map((hit) =>
+      toGlobalSearchProviderResult(hit._id!, hit._source, term, groupStreamsEnabled)
+    );
+  }
+
+  const relevantTypes = types.filter((type) => streamTypes.includes(type));
+  if (relevantTypes.length === 0) {
+    return [];
+  }
+
+  if (relevantTypes.includes('stream')) {
+    return hitsWithAccess.map((hit) =>
+      toGlobalSearchProviderResult(hit._id!, hit._source, term, groupStreamsEnabled)
+    );
+  }
+
+  const includeClassicStream = relevantTypes.includes('classic stream');
+  const includeWiredStream = relevantTypes.includes('wired stream');
+  const includeGroupStream = relevantTypes.includes('group stream');
+  const includeStream = ({ _source }: SearchHit<Streams.all.Definition>) => {
+    return (
+      (includeClassicStream && Streams.ClassicStream.Definition.is(_source)) ||
+      (includeWiredStream && Streams.WiredStream.Definition.is(_source)) ||
+      (includeGroupStream && Streams.GroupStream.Definition.is(_source))
+    );
+  };
+
+  return hitsWithAccess
+    .filter(includeStream)
+    .map((hit) => toGlobalSearchProviderResult(hit._id!, hit._source, term, groupStreamsEnabled));
+}
+
+function toGlobalSearchProviderResult(
+  id: string,
+  definition: Streams.all.Definition,
+  term: string,
+  groupStreamsEnabled: boolean
```

Comment:
group streams should likely be filtered out before reaching this function if the feature is not enabled, so we don't need this here. wdyt ?

Reply:
ah nevermind you also have the `find all` case

Reply:
Yeah, that's what I meant. It should all be filtered anyway and changing the `includeGroupStream` doesn't change anything, so is this okay?

Reply:
Maybe there's a misunderstanding but I think we're just checking twice when we don't have to. the scoring ternary will also score a group stream as an ingest stream if group streams are disabled, but it technically cannot happen because we're guaranteed group stream are enabled if we have a group stream in that function

Reply:
Aaaah I _think_ I get what you mean now.
So you just want to remove passing the flag to this function and using it here, since if a Group stream reached this far it's because the flag is enabled (based on the filter I added above based on your other comment).

If yes, that makes sense to me so I'll just clean up this function and hopefully we're on the same page 😁